commands/command_migrate_import.go

```diff
@@ -202,3 +283,98 @@ func trackedToBlob(db *odb.ObjectDatabase, patterns *tools.OrderedSet) ([]byte,
 		Size:     int64(attrs.Len()),
 	})
 }
+
+// rewriteTree replaces the blob at the provided path within the given tree with
+// a git lfs pointer. It will recursively rewrite any subtrees along the path to the
+// blob.
+func rewriteTree(db *odb.ObjectDatabase, root []byte, path string) ([]byte, error) {
+	tree, err := db.Tree(root)
+	if err != nil {
+		return nil, err
+	}
+
+	splits := strings.SplitN(path, "/", 2)
+
+	switch len(splits) {
+	case 1:
+		// The path points to an entry at the root of this tree, so it must be a blob.
+		// Try to replace this blob with a git lfs pointer.
+		index := findEntry(tree, splits[0])
+		if index < 0 {
+			return nil, errors.Errorf("unable to find entry %s in tree", splits[0])
+		}
+
+		blobEntry := tree.Entries[index]
+		blob, err := db.Blob(blobEntry.Oid)
+		if err != nil {
+			return nil, err
+		}
+
+		var buf bytes.Buffer
+
+		if _, err := clean(lfs.NewGitFilter(cfg), &buf, blob.Contents, blobEntry.Name, blob.Size); err != nil {
+			return nil, err
+		}
+
+		newBlob := odb.Blob{
+			Contents: &buf,
+			Size:     int64(buf.Len()),
+		}
+
+		newOid, err := db.WriteBlob(&newBlob)
+		if err != nil {
+			return nil, err
+		}
+
+		newEntry := odb.TreeEntry{
+			Name:     splits[0],
+			Filemode: blobEntry.Filemode,
+			Oid:      newOid,
+		}
+		tree = tree.Merge(&newEntry)
```

Comment:
Nit; let's inline `newEntry` as well, to keep the current scope as narrow as possible.